go/registry/api/runtime.go

```diff
@@ -316,6 +341,56 @@ type Runtime struct { // nolint: maligned
 
 	// Staking stores the runtime's staking-related parameters.
 	Staking RuntimeStakingParameters `json:"staking,omitempty"`
+
+	// GovernanceModel specifies the runtime governance model.
+	GovernanceModel RuntimeGovernanceModel `json:"governance_model"`
+}
+
+// RuntimeGovernanceModel specifies the runtime governance model.
+type RuntimeGovernanceModel uint8
+
+const (
+	GovernanceInvalid   RuntimeGovernanceModel = 0
+	GovernanceEntity    RuntimeGovernanceModel = 1
+	GovernanceRuntime   RuntimeGovernanceModel = 2
+	GovernanceConsensus RuntimeGovernanceModel = 3
+
+	gmInvalid   = "invalid"
+	gmEntity    = "entity"
+	gmRuntime   = "runtime"
+	gmConsensus = "consensus"
+)
+
+// String returns a string representation of a runtime governance model.
+func (gm RuntimeGovernanceModel) String() string {
+	switch gm {
+	case GovernanceInvalid:
+		return gmInvalid
+	case GovernanceEntity:
+		return gmEntity
+	case GovernanceRuntime:
+		return gmRuntime
+	case GovernanceConsensus:
+		return gmConsensus
+	default:
+		return "[unsupported runtime governance model]"
+	}
+}
+
+// FromString deserializes a string into a RuntimeGovernanceModel.
+func (gm *RuntimeGovernanceModel) FromString(str string) error {
```

Comment:
Instead of this implement `MarshalText` and `UnmarshalText` so things like JSON serialization will work as expected.